internal/cmd/login_test.go

```diff
@@ -0,0 +1,191 @@
+package cmd
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/Netflix/go-expect"
+	"github.com/creack/pty"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/hinshun/vt10x"
+	"golang.org/x/crypto/acme/autocert"
+	"golang.org/x/sync/errgroup"
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/assert/opt"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	"github.com/infrahq/infra/api"
+	"github.com/infrahq/infra/internal/server"
+	"github.com/infrahq/infra/uid"
+)
+
+func TestLoginCmd(t *testing.T) {
+	dir := t.TempDir()
+	t.Setenv("HOME", dir)
+	t.Setenv("USERPROFILE", dir) // Windows
+	// k8s.io/tools/clientcmd reads HOME at import time, so this must be patched too
+	kubeConfigPath := filepath.Join(dir, "kube.config")
+	t.Setenv("KUBECONFIG", kubeConfigPath)
+
+	opts := defaultServerOptions(dir)
+	opts.Addr = server.ListenerOptions{HTTPS: "127.0.0.1:0", HTTP: "127.0.0.1:0"}
+	srv, err := server.New(opts)
+	assert.NilError(t, err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	setupCertManager(t, opts.TLSCache, srv.Addrs.HTTPS.String())
+	go func() {
+		err := srv.Run(ctx)
+		assert.Check(t, err)
+	}()
+
+	runStep(t, "first login prompts for setup", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(ctx)
+		t.Cleanup(cancel)
+
+		console := newConsole(t)
+		ctx = PatchCLIWithPTY(ctx, console.Tty())
+
+		g, ctx := errgroup.WithContext(ctx)
+		g.Go(func() error {
+			// TODO: remove --skip-tls-verify
+			return Run(ctx, "login", srv.Addrs.HTTPS.String(), "--skip-tls-verify")
+		})
+
+		exp := expector{console: console}
+		exp.ExpectString(t, "Username:")
+		exp.Send(t, "admin\n")
+		exp.ExpectString(t, "Password")
+		exp.Send(t, "password\n")
+		exp.ExpectString(t, "Confirm")
+		exp.Send(t, "password\n")
+		exp.ExpectString(t, "Logged in as")
```

Comment:
The test was failing here in CI. I've increased the timeout from 500ms to 2s to see if that helps.

Reply:
Ok, changing to 2s seems to fix it. I guess CI is quite a bit slower because we use `-race`, and the CPU/disk is probably shared with other processes.

Reply:
I spoke too soon. It's flaked again with the 2 second timeout!  

I'm not sure why it's so much slower in CI, but I'll try increasing this further to see if it can be reliable. It's possible there is something else that is causing the flake.

Reply:
I increased the timeout to 20 seconds in CI, but kept it to 1 second for local dev. Hopefully that works.